ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchAlertsTool.java

```diff
@@ -0,0 +1,169 @@
+                @SuppressWarnings("unchecked")
+                List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+
+        // build out the request
+        final String tableSortOrder = parameters.containsKey("sortOrder") ? parameters.get("sortOrder") : "asc";
+        final String tableSortString = parameters.containsKey("sortString") ? parameters.get("sortString") : "monitor_name.keyword";
+        final int tableSize = parameters.containsKey("size") ? Integer.parseInt(parameters.get("size")) : 20;
+        final String searchString = parameters.containsKey("searchString") ? parameters.get("searchString") : null;
+
+        // not exposing "missing" or "startIndex" from the table, using defaults of null/0, respectively
+        final Table table = new Table(tableSortOrder, tableSortString, null, tableSize, 0, searchString);
+
+        final String severityLevel = parameters.containsKey("severityLevel") ? parameters.get("severityLevel") : "ALL";
+        final String alertState = parameters.containsKey("alertState") ? parameters.get("alertState") : "ALL";
+        final String monitorId = parameters.containsKey("monitorId") ? parameters.get("monitorId") : null;
+        final String alertIndex = parameters.containsKey("alertIndex") ? parameters.get("alertIndex") : null;
+        @SuppressWarnings("unchecked")
+        final List<String> monitorIds = parameters.containsKey("monitorIds")
+            ? gson.fromJson(parameters.get("monitorIds"), List.class)
+            : null;
+        @SuppressWarnings("unchecked")
+        final List<String> workflowIds = parameters.containsKey("workflowIds")
+            ? gson.fromJson(parameters.get("workflowIds"), List.class)
+            : null;
+        @SuppressWarnings("unchecked")
+        final List<String> alertIds = parameters.containsKey("alertIds") ? gson.fromJson(parameters.get("alertIds"), List.class) : null;
+
+        GetAlertsRequest getAlertsRequest = new GetAlertsRequest(
+            table,
+            severityLevel,
+            alertState,
+            monitorId,
+            alertIndex,
+            monitorIds,
+            workflowIds,
+            alertIds
+        );
+
+        // create response listener
+        ActionListener<GetAlertsResponse> getAlertsListener = ActionListener.<GetAlertsResponse>wrap(response -> {
+            listener.onResponse((T) response);
+        }, e -> { listener.onFailure(e); });
+
+        // execute the search
+        AlertingPluginInterface.INSTANCE.getAlerts((NodeClient) client, getAlertsRequest, getAlertsListener);
+    }
+
+    @Override
+    public boolean validate(Map<String, String> parameters) {
+        if (parameters == null) {
+            return false;
```

Comment:
If parameters is null can it be initialized to empty map instead of validating it to false? Ideally validate should be false if tool will not be able to handle the request.

Reply:
This is currently following the pattern of existing tools, I have no preference. Prefer to leave as-is until there is a final consistent way parameters should be handled, and `validate()` can be updated for all of the tools. What do you think?

Reply:
I would prefer to have tool specific implementation, given that alerting tool can work even if none of the parameters are provided, I would prefer not to have this check. I am planning to build an AbstractTool class with default validate implementation to return true, with that change AlertingTool need not then override `validate()` method.

Reply:
Ok, I'll change to default to `true` for everything for now, in the future we may remove and fall back to the default when that is available.